### ISSUE TITLE
Potential react component bug (demo project included)

### ISSUE THREAD

**Issue by Tredis** [necolas/react-native-web#584]
My team is trying to use this fantastic library for our principal commercial app. It has been really great so far, but have run into the following issue:

**Do you want to request a *feature* or report a *bug*?**
Reporting a potential bug

**What is the current behavior?**
React component libraries do not work as intended in browser

**If the current behavior is a bug, please provide the steps to reproduce and
minimal demo: https://github.com/Tredis/RNWdemo
instructions and details in the readme

**What is the expected behavior?**
React libraries to work as expected

**Environment (include versions). Did this work in previous versions?**
These issues are observed across multiple versions, including 0.0.112. We tried to upgrade to 0.0.116, but main app gets errors, so we reverted back to 0.0.112.

* OS: OSX ElCapital & OSX Sierra
* Browser: All
* React Native for Web (version): 0.0.112
* React (version): 15.6.1

**Comment by necolas** [comment]
I have no idea what "React component libraries do not work as intended in browser" means.

(Please can you use Glitch to host functioning examples. Much easier than having to install an entire project and figure out what is going on.)

**Comment by necolas** [comment]
I tried `react-datetime` and can't reproduce any issue: https://natural-mallet.glitch.me/

Did you forget to manually include CSS files for those dependencies? It's mentioned [here](https://github.com/YouCanBookMe/react-datetime#usage) but your demo app doesn't include them.

**Comment by Tredis** [comment]
Thanks for the quick reply. I could make a Glitch project, but I have never used Glitch before and have already invested significant time in creating a standalone project with a well documented readme describing expected vs. actual behavior including screenshots. I respectfully ask that you give the demo project (at least the readme) another look as it documents the problems fairly well. I feel it may be a very simple thing we are missing (maybe babel related) either in how we include the library or in the webpack.config. You may not even need to run it to realize what's wrong.

For convenience, I am pasting the readme that includes all details including screenshots here:

### Library 1: [react-datetime](https://github.com/YouCanBookMe/react-datetime)
React-datetime is a UI widget which allows a user to pick from a calendar and set a time of day. This is the widget you will see when you run the demo, and you will notice several features are broken compared to when this widget is running without react-native-web. See [this codepen](https://codepen.io/simeg/pen/mEmQmP) for a fully working demo.

Problems:

* calendar locked in dropped state
* numbers outside of current month are not grayed
* cells do not highlight on mouseover
* time choosers are not aligned
* much of the styling is lost
 
Expected react-datetime componant:

Calendar tab                                                                       |  Time tab
:---------------------------------------------------------------------------------:|:---------------------------------------------------------------------------------:
![Alt text](https://user-images.githubusercontent.com/22584383/28801712-c776bc4e-7620-11e7-985f-dbf80dfb9433.png "Expected calendar tab")  |  ![Alt text](https://user-images.githubusercontent.com/22584383/28801715-cc35c1e4-7620-11e7-85ca-30794fc8b16c.png "Expected time tab")

Actual react-datetime componant:

Calendar tab                                                                       |  Time tab
:---------------------------------------------------------------------------------:|:---------------------------------------------------------------------------------:
![Alt text](https://user-images.githubusercontent.com/22584383/28801720-d4a28fd8-7620-11e7-8b5f-4847b0ea4c21.png "Actual calendar tab")      |  ![Alt text](https://user-images.githubusercontent.com/22584383/28801732-dcedeaf2-7620-11e7-98f9-ed5eb5a399dd.png "Actual time tab")

### Library 2: [react-dropdown](https://github.com/fraserxu/react-dropdown)
React-dropdown is a drop down UI widget. [Fully working demo](http://fraserxu.me/react-dropdown/). This widget is not even showing up in the demo project.

### Library 3: [input-moment](https://github.com/wangzuo/input-moment)
[Fully working demo](http://wangzuo.github.io/input-moment/). This component is currently commented out at line 21 or file source/components/App/index.js. Remove the comment to observe the error.

### Library 4: [react-datetime-calendar](https://github.com/deepreact/react-datetime-calendar)
react-datetime-calendar errors out, but can be fixed by changing "file" to "file-loader" in node_modules, at the top of lib/DateTimePicker.js but once it displays, it has similar visual issues as react-datetime.

Expected react-datetime-calendar component                                                  |  Actual react-datetime-calendar component
:------------------------------------------------------------------------------------------:|:----------------------------------------------------------------------------------------:
![Alt text](https://user-images.githubusercontent.com/22584383/28803278-6368223e-7629-11e7-9985-342aefbf3b75.png "Expected react-datetime-calendar")|![Alt text](https://user-images.githubusercontent.com/22584383/28803283-65ce56d8-7629-11e7-9c60-2a8e4b9f3221.png "Actual react-datetime-calendar")

## Conclusion
Given the errors, I believe that we may be doing something wrong with babel/webpack. I just don't know what.

**Comment by Tredis** [comment]
Sorry, I didn't see you replied. I will check the CSS files instructions you mentioned

**Comment by Tredis** [comment]
Indeed, I did miss that CSS file inclusion for react-datetime. None of the other 3 libraries have instructions like that though, and I haven't seen any other react components that require you to do that before.
Can you see a more general solution for the other react components? I still can help but feel there's still some simple babel issue we're missing. Does the webpack.config file look ok? Should we be using createDOMElement in source/component/App/index.js if we only want to use default styles?

**Comment by necolas** [comment]
Components built directly on react-dom are usually always precompiled. If they need webpack loaders for CSS or images they would normally document that. You won't need to do anything in 'createDOMElement' for default behavior

**Comment by Tredis** [comment]
That is what I thought too. I have simplified the demo repo. Only 2 react libs remain, and neither require any special CSS or loaders.

react-dropdown component does not show up on screen
and
react-datetime-calendar component shows up, but it is incomplete as per the screenshots.

Any help on these would be greatly appreciated. Again I feel it could be a babel issue.

**Comment by necolas** [comment]
I think you need to ask the people maintaining those projects

**Comment by Tredis** [comment]
I have confirmed these work in another pure react app. (besides, react-dropdown has almost as many downloads as RNW at 24k/mth, so it's pretty solid)
The fact that there are many other libraries that exhibit the same issues tells me I might be incorrectly implementing RNW. Can you at least confirm my webpack.config looks ok?

```
const webpack = require('webpack');

module.exports = {
    entry: ['babel-polyfill', './index.web.js'], //need babel-polyfill here for sagas

    output: {
        filename: 'bundle.js'
    },
    module: {
        loaders: [
            {
                test: /\.js$/,
                exclude: [
                    /node_modules_gitsync/,
                    /node_modules\/react-dropdown/,
                    /node_modules\/react-native-web/,
                ],
                include: [/node_modules/, /source/],
                loader: 'babel-loader',
                query: {
                    presets: ['env', 'react-native'],
                }
            }, {
                test: /\.(jpe?g|png|gif|svg|ico)$/i,
                loader: 'file-loader',
                  options: {
                    name: '[path][name].[hash].[ext]',
                  },
            }
        ]
    },
    resolve: {
        alias: {
            "react-native": "react-native-web",
            'react-navigation': 'react-navigation/lib/react-navigation.js',
        },
        extensions: [ '.web.js', '.js' ]
    }
};
```

**Comment by necolas** [comment]
css file -> https://github.com/fraserxu/react-dropdown/blob/master/style.css

**Comment by Tredis** [comment]
Thanks. I guess some libraries don't document the need to include the css file? I tried to include the raw link (https://raw.githubusercontent.com/fraserxu/react-dropdown/master/style.css) to the css file you linked as you did in the glitch example, but it still didn't work. I noticed you used a rawgit link? What is that? And where can I find it for this library?

And where can I find some documentation for this, so I can stop bothering you? :)
